### ISSUE TITLE
"Text Draw Speed" causes Unexpected Behavior with escape code "\006\mask"

### ISSUE THREAD

**Issue by 8bitKittyKat** [chrismaltby/gb-studio#1491]
**Describe the bug**
I'm using the `\006\020` escape sequence in a text box to require the player to press the A button to scroll to the next line in a dialog box, but changing the text drawing speed from the default of 1 causes unexpected behavior. If drawing speed is set to 1 it works as expected, but any other text drawing speed requires multiple keypresses from the player, and sometimes it will appear to refuse to continue at all.

**To Reproduce**
1. Add a basic `Display Dialog` event, then enter the following sequence of alphanumeric characters, which are each separated by the escape code `\006\020`, into its text window
>a\006\020b\006\020c\006\020d\006\020e\006\020f\006\020g\006\020h\006\020i\006\020j\006\020k\006\020l\006\020m\006\020n\006\020o\006\020p\006\020q\006\020r\006\020
>s\006\020t\006\020u\006\020v\006\020w\006\020x\006\020y\006\020z\006\0201\006\0202\006\0203\006\0204\006\0205\006\0206\006\0207\006\0208\006\0209\006\0200

2. Add the `Text Animation Speed` event before the `Display Dialog`, and leave the setting at 1.

3. Test the game, and when you activate the dialog you should see each character in the text box draw one at a time with a single press of the A button.

4. Go back to GB Studio and change `Text Draw Speed` in the `Text Animation Speed` event to any number from 2 - 5.

5. Test the game again, and when you activate the dialog this time, each character will require multiple keypresses to draw each consecutive character, and at some point it will appear to stop responding to input.

**Expected behavior**
The window should draw each character after a single keypress regardless of text speed since it's only drawing a single character at a time but instead slower text speeds cause long delays or multiple keypresses to get the next character to draw.

**Platform (please complete the following information):**
 - OS: Arch Linux x86_64
 - App version: 4.0.1

**Additional context**
The steps to reproduce involve code that differs from what I'm using in my project, since that involves multiple lines of text before player input is required, but the steps I provided here serve to show that there's something inherently wrong occurring between the escape code for requiring player input to continue text drawing, and changing text drawing speed.

**Comment by 8bitKittyKat** [comment]
I may have discovered a fix for this issue, but I have only done a minimal amount of testing.  I ejected the engine and made edits to the section of `ui.c` that determines what happens when it encounters control codes.
```
            case 0x06:
                current_text_speed = ui_time_masks[1];
```

and

```
            default:
                if (ui_print_render(*ui_text_ptr)) {
                    ui_set_tile(ui_dest_ptr, ui_prev_tile, ui_prev_tile_bank);
                    if (vwf_direction == UI_PRINT_LEFTTORIGHT)  ui_dest_ptr++; else ui_dest_ptr--;
                }
                if (vwf_current_offset) ui_set_tile(ui_dest_ptr, ui_current_tile, ui_current_tile_bank);
                ui_text_ptr++;
                current_text_speed = ui_time_masks[text_draw_speed];
                return TRUE;
```

I just added the two `current_text_speed` lines above.  The first sets the text draw speed to 1 when it sees the input control code, and then it resets it back to the original draw speed when it gets to the end of the control code loop.  I have very limited experience with C and this was achieved through some experimentation but seems to be working.

**Comment by chrismaltby** [comment]
Hi @8bitKittyKat thanks for raising this, I've implemented a fix based on what you've suggested in 4.0.2 which I've released today, hopefully that works a bit better for you now. Sounds like @untoxa has a more elegant longer term fix in mind but I think this will work for now. Thanks again

**Comment by 8bitKittyKat** [comment]
Just downloaded 4.0.2 and it seems to be working well, thanks for implementing this fix in such short time!  Going to mark this as closed.
